File: Lab1.py
import requests
import re
Urls = 'http://www.csd.tsu.ru/'
LimitDeep = 10
html = requests.get(Urls)
Emails = []
VisUrl = []
def Pages(html):
	Output = re.findall(r'\b[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+(?<![.])', html.text)
	Flag = 0
	for i in range(len(Output)):
		for j in range(len(Emails)):
			if Output[i] == Emails[j]:
					Flag = 1
		if Flag == 0:
			Emails.append(Output[i])
		Flag = 0
	Output = []
def Survey(html, deep):
	VisUrl.append(html.url)
	deep += 1
	d = deep
	Pages(html)
	New = Urls + '([A-z0-9-/+=&?_$]+)'
	Reference = re.findall(r'<a href=[\'\"]'+ New + '[\'\"]>', html.text)
	if deep < LimitDeep:
		Check = 0
		for i in range(len(Reference)):
			for j in range(len(VisUrl)):
				if Urls + Reference[i] == VisUrl[j]:
					Check = 1
			if Check == 0:
				rg = requests.get(Urls + Reference[i])
				Survey(rg, deep)
			Check = 0
		deep = d
Survey(html,0)	
print(Emails)

File: lab1.js
var request = require('request');
var address = 'http://www.csd.tsu.ru';

var VisUrl=[];
var Emails=[];
var EmailsAmount = 0;

var emailRegex = /\b[a-zA-Z0-9_.+-]+@[a-zA-Z0-9-]+.[a-zA-Z0-9-.]+/ig;
var linkRegex = /<a href=\"(\/[-+\w:\/#@$.]*)/ig;
var limitDeep = 10;

start(address, 0);

function start(url, iter)
{
	iter=iter+1;
    request(url, function (err,res,body) 
    {
        if (err) {
           //throw err;
        }
        else {
            var pageEmails=body.match(emailRegex);
            if(typeof(pageEmails)!=null) {
                pageEmails=getOnly(pageEmails);
            }
            if(pageEmails) {
                for (i = 0; i < pageEmails.length; i++) {
                    if (Emails.indexOf(pageEmails[i]) == -1) {
                        Emails.push(pageEmails[i]);
                    }
                }
            }

            links=body.match(linkRegex);
            if (typeof(links) != null && links) {
                links=getOnly(links);
                for(i=0; i<links.length; i++)
                {
                    if (VisUrl.indexOf(links[i]) == -1 && iter<limitDeep) {
                        page=links[i].split('href="')[1];
                        start(address + page, iter);
                        VisUrl.push(links[i]);
                        //console.log("VisUrl: " + links[i]);
                    }
                }   
            }
        }
        
        if (EmailsAmount < Emails.length) 
        {
            console.log("Теперь надено " + Emails.length + ": " + Emails + "\n ---------");
            EmailsAmount = Emails.length;
        }
    });

    
}

function getOnly(e) {
  var result = [];
  if (typeof e != 'undefined') {
    for (i = 0; i < e.length; i++){
      if (result.indexOf(e[i]) == -1) {
          result.push(e[i])
      }
    }
  }

  return result;
}